tests/unit/core/test_utils.js

```diff
@@ -957,3 +972,106 @@ describe('getClientCompatibility', () => {
     });
   });
 });
+
+describe('getCategoryColor', () => {
+  it('throws if category is false-y', () => {
+    expect(() => {
+      getCategoryColor(null);
+    }).toThrowError('category is required');
+  });
+
+  it('all valid addonTypes are in CATEGORY_COLORS', () => {
+    expect(() => {
+      validAddonTypes.forEach((addonType) => {
+        const category = { id: 2, type: addonType };
+
+        getCategoryColor(category);
+      });
+    }).not.toThrowError('not found in CATEGORY_COLORS');
+  });
+
+  it('throws on unrecognised addonType', () => {
+    expect(() => {
+      getCategoryColor({ id: 5, type: 'NOT_A_REAL_TYPE' });
+    }).toThrowError(
+      'addonType "NOT_A_REAL_TYPE" not found in CATEGORY_COLORS');
+  });
+
+  it('deals with high category IDs', () => {
+    for (let i = 750; i < 800; i++) {
+      const category = { id: i, type: ADDON_TYPE_THEME };
+      const categoryColor = getCategoryColor(category);
+
+      expect(categoryColor)
+        .toBeLessThanOrEqual(CATEGORY_COLORS[ADDON_TYPE_THEME]);
+      expect(categoryColor).toBeGreaterThanOrEqual(1);
+    }
+  });
+
+  it('has a different number of colors for different addonTypes', () => {
+    const category = { id: 11, type: ADDON_TYPE_EXTENSION };
+    const categoryColor = getCategoryColor(category);
+
+    expect(categoryColor).toEqual(1);
+    expect(categoryColor).toBeLessThan(11);
```

Comment:
This assertion seems unnecessary as long as you are using `toEqual()` right before it